WcaOnRails/app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -257,13 +257,16 @@ def index_for_group_type
   def create
     user_id = params.require(:userId)
     group_id = params[:groupId] || UserGroup.find_by(group_type: params.require(:groupType)).id
-    group = UserGroup.find(group_id)
 
     if group_id.is_a?(String) && group_id.include?("_") # Temporary hack to support some old system roles, will be removed once all roles are
       # migrated to the new system.
       group_type = group_id_of_old_system_to_group_type(group_id)
       original_group_id = group_id.split("_").last
-      if group_type == UserGroup.group_types[:councils]
+      unless current_user.can_edit_team?(original_group_id)
+        render json: {}, status: 401
+        return
+      end
```

Comment:
Depending on what your frontend expects (i.e. wheter a blank non-JSON body that contains absolutely nothing (as opposed to a JSON string representing nothing) is fine or not) you may replace this with `return head :unauthorized`

Reply:
Even if you somehow need an empty JSON (in which case you should reconsider your frontend design) you can still simplify this to one line with `return render json: {}, status: :unauthorized`

Reply:
`return head :unauthorized` looks good. This is currently not used in frontend because this happens in very rare case - like if a leader access the leader panel with the leader access and while they are in leader page, their leader access got removed. The UI is currently not handling the error case but will handle it in future.

I've changed to `return head :unauthorized`.